src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,512 @@
+        Type returnType = isSearchingForTargetType ? t : currentType;
+        for (Executable executable : executableList) {
+          if (executable instanceof Constructor
+              || (executable instanceof Method
+                  && returnType.isAssignableFrom(
+                      Type.forClass(((Method) executable).getReturnType())))) {
+
+            // Obtain the input types and output type of the executable.
+            List<Type> inputTypeList = classArrayToTypeList(executable.getParameterTypes());
+            // If the executable is a non-static method, add the receiver type to
+            // the front of the input type list.
+            if (executable instanceof Method && !Modifier.isStatic(executable.getModifiers())) {
+              inputTypeList.add(0, new NonParameterizedType(currentClass));
+            }
+            TypeTuple inputTypes = new TypeTuple(inputTypeList);
+            CallableOperation callableOperation =
+                executable instanceof Constructor
+                    ? new ConstructorCall((Constructor<?>) executable)
+                    : new MethodCall((Method) executable);
+            NonParameterizedType declaringType = new NonParameterizedType(currentClass);
+            TypedOperation typedClassOperation =
+                new TypedClassOperation(callableOperation, declaringType, inputTypes, returnType);
+
+            // Add the method call to the result.
+            result.add(typedClassOperation);
+            producerParameterTypes.addAll(inputTypeList);
+          }
+          processed.add(currentType);
+          // Add the parameter types of the current method to the workList.
+          // This allows the search for methods that can produce these parameter types,
+          // thereby creating the sequences of methods needed to generate the input types
+          // for methods that lead to the generation of the specified type.
+          workList.addAll(producerParameterTypes);
+        }
+      }
+      isSearchingForTargetType = false;
+    }
+
+    // TODO: Reverse the result may improve the quality of the generated tests.
+    // Producer methods are added to the list in the order they are needed. However, objects are
+    // often built up from the simplest types. Reversing the result may help generate
+    // basic types first, leading to the generation of more complex types within fewer tests.
+    // This needs to be looked into further.
+    Collections.reverse(result);
+
+    return new LinkedHashSet<>(result);
+  }
+
+  /**
+   * Given an array of classes, this method converts them into a list of {@code Types}.
+   *
+   * @param classes an array of reflection classes
+   * @return a list of Types
+   */
+  private static List<Type> classArrayToTypeList(Class<?>[] classes) {
+    return CollectionsPlume.mapList(Type::forClass, classes);
+  }
+
+  /**
+   * This method creates a new sequence for the given {@code TypedOperation}. The method iteratively
```

Comment:
Randoop contains other code that creates a new sequence for a given TypedOperation and has a specification very similar to this one.  Please reference that code and explain how this differs.